var/spack/repos/builtin/packages/repeatmodeler/package.py

```diff
@@ -44,6 +44,46 @@ class Repeatmodeler(Package):
     depends_on("blat", type="run", when="@2.0.4:")
     depends_on("ltr-retriever", type="run", when="@2.0.4:")
 
+    def patch(self):
+        file_list = [
+            "BuildDatabase",
+            "Job.pm",
```

Comment:
This will break 1.0.11 (or rather break it more) - the file does not exist there and the build will error out.
Not sure if it makes sense to just drop 1.0.11.

Reply:
@EbiArnie Do you think ignore_absent=True will be sufficient here?

Reply:
Sorry - this dropped off my radar.
`ignore_absent=True` will probably do the trick.

I'm using a much simpler package.py, though, see below.

My recommendation would be to deprecate 1.0.11, add 2.0.5 and simplify package.py.
I'm only patching the script executables - we don't have to care for wrong shebang lines in the the module files.
And I link the script executables into the bin path to make things work.
For the config, I ignore the configuration script and patch the config file directly.
This makes for a simpler package.py. I personally prefer the simpler version. We have been using this one in production for a while.
If you want to go with my version, I'm happy. 1.0.11 should be added back in, as well as 2.0.4 and probably 2.0.5. Spack policy is to deprecate versions at least for one release before dropping them, so we can't immediately get rid of 1.0.11.
If you prefer what's there at the moment, that's fine too. Ping me and I'll review as-is.

Relevant part of my package.py:
```
    depends_on("perl", type=("build", "run"))
    depends_on("perl-json", type=("build", "run"))
    depends_on("perl-uri", type=("build", "run"))
    depends_on("perl-libwww-perl", type=("build", "run"))
    depends_on("perl-file-which", type=("build", "run"), when="@2.0.1:")
    depends_on("perl-devel-size", type=("build", "run"), when="@2.0.1:")
    depends_on("repeatmasker", type="run")
    depends_on("recon +repeatmasker", type="run")
    depends_on("repeatscout", type="run")
    depends_on("trf", type="run")
    depends_on("nseg", type="run")
    depends_on("ncbi-rmblastn", type="run")
    # "optional" dependencies that it still wants
    depends_on("cdhit", type="run", when="@2.0.1:")
    depends_on("genometools", type="run", when="@2.0.1:")
    depends_on("mafft", type="run", when="@2.0.1:")
    depends_on("ninja-phylogeny", type="run", when="@2.0.1:")
    depends_on("ltr-retriever", type="run", when="@2.0.1:")

    def install(self, spec, prefix):
        cdhitdir = spec["cdhit"].prefix.bin
        genometoolsdir = spec["genometools"].prefix.bin
        ltrdir = spec["ltr-retriever"].prefix.bin
        mafftdir = spec["mafft"].prefix.bin
        ninjadir = spec["ninja-phylogeny"].prefix.bin
        recondir = spec["recon"].prefix.bin
        repeatmaskerdir = spec["repeatmasker"].prefix.lib
        rmblastdir = spec["ncbi-rmblastn"].prefix.bin
        rscoutdir = spec["repeatscout"].prefix.bin
        trfbin = os.path.join(spec["trf"].prefix.bin, "trf")
        cfgfile = "RepModelConfig.pm"
        filter_file("REPLACE_CDHIT", cdhitdir, cfgfile)
        filter_file("REPLACE_GENOMETOOLS", genometoolsdir, cfgfile)
        filter_file("REPLACE_LTR", ltrdir, cfgfile)
        filter_file("REPLACE_MAFFT", mafftdir, cfgfile)
        filter_file("REPLACE_NINJA", ninjadir, cfgfile)
        filter_file("REPLACE_RECON", recondir, cfgfile)
        filter_file("REPLACE_REPEATMASKER", repeatmaskerdir, cfgfile)
        filter_file("REPLACE_RMBLAST", rmblastdir, cfgfile)
        filter_file("REPLACE_RSCOUT", rscoutdir, cfgfile)
        filter_file("REPLACE_TRF", trfbin, cfgfile)

        executables = ["BuildDatabase", "LTRPipeline", "Refiner", "RepeatClassifier", "TRFMask", "RepeatModeler"]

        for file in executables:
            filter_file("#!.*", "#!/usr/bin/env perl", file)

        install_tree(".", prefix.lib)

        mkdirp(prefix.bin)

        for file in executables:
            os.symlink(join_path(prefix.lib, file),
                       join_path(prefix.bin, file))

```